apps/frontend/src/libs/hooks/use-infinity-scroll/use-infinity-scroll.hook.ts

```diff
@@ -0,0 +1,46 @@
+import { useCallback, useMemo, useState } from "~/libs/hooks/hooks.js";
+
+import { FIRST_PAGE, PAGE_INCREMENT } from "./libs/constants/constants.js";
+
+type UseInfiniteScroll = (parameters: {
+	pageSize: number;
+	totalItemsCount: number;
+}) => {
+	hasNextPage: boolean;
+	onPageChange: () => void;
+	onPageReset: () => void;
+	page: number;
+	pageSize: number;
+};
+
+const useInfiniteScroll: UseInfiniteScroll = ({
+	pageSize,
+	totalItemsCount,
+}) => {
+	const [page, setPage] = useState<number>(FIRST_PAGE);
+
+	const hasNextPage = useMemo(
+		() => (page + PAGE_INCREMENT) * pageSize < totalItemsCount,
```

Comment:
Not sure if this "formula" is correct.

For example, let assume that we have 15 items in total. Page size is 10. Let the page be FIRST_PAGE (i.e. 0).
Then hasNextPage will be true (that's correct): (0+1)*10 < 15.
But if we load first page, then it wil be 1, and hasNextPage now will be false: (1+1)*10 >= 15 -- that's seems to be not right, cause we have 5 extra items, which should be on the next page, so hasNextPage should be true.

Maybe I don't fully understand the flow, but pay attention to this.

Reply:
The numbering starts from 0, so page 1 will be the second. On page 0 it will load 10 items, on page 1 - the next 5

Reply:
Oh, yeah, I took it into account. I just got confused while writing example. Your formula is correct :)